src/Nethermind/Nethermind.Consensus/Producers/BuildBlocksOnlyWhenNotProcessing.cs

```diff
@@ -100,6 +100,7 @@ private void InvokeTriggerBlockProduction(BlockProductionEventArgs e)
                 if (_logger.IsDebug) _logger.Debug($"Delaying producing block, chain not processed yet. BlockProcessingQueue count {_blockProcessingQueue.Count}.");
                 await Task.Delay(ChainNotYetProcessedMillisecondsDelay, cancellationToken);
             }
+            cancellationToken.ThrowIfCancellationRequested();
```

Comment:
So, the test expect to receive a `OperationCancelledException` from line 101. But sometimes the loop exit without throwing an exception which is the reason why the test randomly fail.

Reply:
If it is `sometimes`, when did got thrown when it actually did?

Reply:
Wouldn't it be better to remove `&& !cancellationToken.IsCancellationRequested` part of the condition then? We could probably also remove the condition in line 104 then?

Reply:
Done